rust/otap-dataflow/crates/pdata/src/views/otlp/bytes/metrics.rs

```diff
@@ -0,0 +1,433 @@
+            SCOPE_METRICS_METRICS => self.first_metric.get(),
+            _ => return None,
+        };
+
+        from_option_nonzero_range_to_primitive(range)
+    }
+
+    fn set_field_range(&self, field_num: u64, wire_type: u64, start: usize, end: usize) {
+        let range = match to_nonzero_range(start, end) {
+            Some(range) => range,
+            None => return,
+        };
+
+        if wire_type == wire_types::LEN {
+            match field_num {
+                SCOPE_METRICS_SCOPE => self.scope.set(Some(range)),
+                SCOPE_METRICS_SCHEMA_URL => self.schema_url.set(Some(range)),
+                SCOPE_METRICS_METRICS => {
+                    if self.first_metric.get().is_none() {
+                        self.first_metric.set(Some(range))
+                    }
+                }
+                _ => { /* ignore */ }
+            }
+        }
+    }
+}
+
+/// Implementation of [`MetricView`] backed by protobuf serialized `Metric` message
+pub struct RawMetric<'a> {
+    byte_parser: ProtoBytesParser<'a, MetricFieldRanges>,
+}
+
+/// Known offsets within byte buffer for fields in the `Metric` message
+pub struct MetricFieldRanges {
+    name: Cell<Option<(NonZeroUsize, NonZeroUsize)>>,
+    description: Cell<Option<(NonZeroUsize, NonZeroUsize)>>,
+    unit: Cell<Option<(NonZeroUsize, NonZeroUsize)>>,
+    data: Cell<Option<(NonZeroUsize, NonZeroUsize)>>,
+    first_metadata: Cell<Option<(NonZeroUsize, NonZeroUsize)>>,
+}
+
+impl FieldRanges for MetricFieldRanges {
+    fn new() -> Self {
+        Self {
+            name: Cell::new(None),
+            description: Cell::new(None),
+            unit: Cell::new(None),
+            data: Cell::new(None),
+            first_metadata: Cell::new(None),
+        }
+    }
+
+    fn get_field_range(&self, field_num: u64) -> Option<(usize, usize)> {
+        let range = match field_num {
+            METRIC_NAME => self.name.get(),
+            METRIC_DESCRIPTION => self.description.get(),
+            METRIC_UNIT => self.unit.get(),
+            METRIC_GAUGE
+            | METRIC_SUM
+            | METRIC_HISTOGRAM
+            | METRIC_EXPONENTIAL_HISTOGRAM
+            | METRIC_SUMMARY => self.data.get(),
+            METRIC_METADATA => self.first_metadata.get(),
+            _ => return None,
+        };
+
+        from_option_nonzero_range_to_primitive(range)
+    }
+
+    fn set_field_range(&self, field_num: u64, wire_type: u64, start: usize, end: usize) {
+        if wire_type == wire_types::LEN {
+            let range = match to_nonzero_range(start, end) {
+                Some(range) => range,
+                None => return,
+            };
+
+            match field_num {
+                METRIC_NAME => self.name.set(Some(range)),
+                METRIC_DESCRIPTION => self.description.set(Some(range)),
+                METRIC_UNIT => self.unit.set(Some(range)),
+                METRIC_GAUGE | METRIC_SUM | METRIC_HISTOGRAM | METRIC_EXPONENTIAL_HISTOGRAM => {
```

Comment:
Include `METRIC_SUMMARY` as well?

Reply:
Thanks @utpilla you're right. Made this change